### ISSUE TITLE
sydtest-hspec: `shouldBe` failure output

### ISSUE THREAD

**Issue by sgillespie** [NorfairKing/sydtest#103]
When using `fromHspec` and getting a comparison failure, I expect the failure output to be the usual helpful diff message:
```
      Expected these values to be equal:
      Actual:   8
      Expected: 6
```

However, what I actually see is:
```
Failures:

    src/Test/Syd/Hspec.hs:68
  ✗ 1 Test.Syd.HspecSpec.exampleHspecSpec.adds 3 and 5 together in io
      Retries: 3 (does not look flaky)
      Contextual "NotEqualButShouldHaveBeenEqualWithDiff \"8\" \"6\" (Just [First \"8\",Second \"6\"])" "test/Test/Syd/HspecSpec.hs:19:44"
```

I'm new to sydtest, so it's possible I'm doing something wrong, although it _looks_ like a bug. I suspect the culplrit is [here](https://github.com/NorfairKing/sydtest/blob/master/sydtest-hspec/src/Test/Syd/Hspec.hs#L146)

**Comment by NorfairKing** [comment]
@sgillespie please show me the code where this is happening :)
PR welcome.

**Comment by sgillespie** [comment]
> @sgillespie please show me the code where this is happening :) PR welcome.

Here's a reproducer: https://github.com/sgillespie/sydtest-reproducer/blob/main/test/Test/HspecSpec.hs

**Comment by NorfairKing** [comment]
@sgillespie I appreciate it, but your code is open-source, right? So I'd like to see it :)

**Comment by sgillespie** [comment]
Absolutely! Although it's a bit of a toy project: https://github.com/sgillespie/obloxious/edit/main/test/Development/Server/Parser/InternalSpec.hs
